examples/src/main/java/com/example/ftclibexamples/MecanumDrivingSample.java

```diff
@@ -1,61 +1,57 @@
 
-    private MecanumDrive driveTrain;
-
-    private MotorEx fL, fR, bL, bR;
-
-    // the motors used here are goBILDA Yellow Jackets, 435 rpm
+    static final boolean FIELD_CENTRIC = false;
 
     @Override
     public void runOpMode() throws InterruptedException {
-
-        fL = new MotorEx(hardwareMap, "frontLeft", MotorEx.GoBILDA.RPM_435);
-        fR = new MotorEx(hardwareMap, "frontRight", MotorEx.GoBILDA.RPM_435);
-        bL = new MotorEx(hardwareMap, "backLeft", MotorEx.GoBILDA.RPM_435);
-        bR = new MotorEx(hardwareMap, "backRight", MotorEx.GoBILDA.RPM_435);
-
-        driveTrain = new MecanumDrive(
-                fL, fR, bL, bR
+        // constructor takes in frontLeft, frontRight, backLeft, backRight motors
+        // IN THAT ORDER
+        MecanumDrive drive = new MecanumDrive(
+            new Motor(hardwareMap, "frontLeft", Motor.GoBILDA.RPM_435),
+            new Motor(hardwareMap, "frontRight", Motor.GoBILDA.RPM_435),
+            new Motor(hardwareMap, "backLeft", Motor.GoBILDA.RPM_435),
+            new Motor(hardwareMap, "backRight", Motor.GoBILDA.RPM_435)
         );
 
-        waitForStart();
-
-        driveWithVector(new Vector2d(12, 3));
-        sleep(1000);
-        driveWithVector(new Vector2d(0, 0));
+        RevIMU imu = new RevIMU(hardwareMap);
+        imu.init();
 
-    }
+        GamepadEx driverOp = new GamepadEx(gamepad1);
 
-    private void driveWithVector(Vector2d vector) {
-        double[] speeds = normalize(new double[]{vector.getX(), vector.getY()});
-        driveTrain.driveRobotCentric(speeds[0], speeds[1], 0);
-    }
+        waitForStart();
 
-    /**
-     * Normalize the wheel speeds if any value is greater than 1
-     */
-    private double[] normalize(double[] wheelSpeeds) {
-        double maxMagnitude = Math.abs(wheelSpeeds[0]);
-        for (int i = 1; i < wheelSpeeds.length; i++) {
-            double temp = Math.abs(wheelSpeeds[i]);
-            if (maxMagnitude < temp) {
-                maxMagnitude = temp;
-            }
-        }
-        if (maxMagnitude > 1.0) {
-            for (int i = 0; i < wheelSpeeds.length; i++) {
-                wheelSpeeds[i] = wheelSpeeds[i] / maxMagnitude;
+        while (!isStopRequested()) {
+
+            if (!FIELD_CENTRIC) {
+                // optional fourth parameter for squared inputs
+                drive.driveRobotCentric(
+                        driverOp.getLeftX(),
+                        driverOp.getLeftY(),
+                        driverOp.getRightX(),
+                        false
+                );
+            } else {
+                // optional fifth parameter for squared inputs
+                drive.driveFieldCentric(
+                        driverOp.getLeftX(),
+                        driverOp.getLeftY(),
+                        driverOp.getRightX(),
+                        imu.getRotation2d().getDegrees(),   // gyro value passed in here must be in degrees
+                        false
+                );
```

Comment:
I think we should indicate what direction each joystick controls. Maybe a diagram like in RR?

Reply:
That doesn't really work for field centric, but I can try my best to come up with something! The diagrams in RR are lit, so I'll see what I can do.

Reply:
Damn I wonder who made those diagrams he must be a cool dude 😎😎